src/Components/Steps/Expenses/Expenses.tsx

```diff
@@ -0,0 +1,323 @@
+  const nextStep = useGoToNextStep('hasExpenses');
+  const expenseOptions = useConfig('expense_options');
+
+  const oneOrMoreDigitsButNotAllZero = /^(?!0+$)\d+$/;
+  const expenseSourceSchema = z.object({
+    expenseSourceName: z.string().min(1),
+    expenseAmount: z.string().regex(oneOrMoreDigitsButNotAllZero),
+  });
+  const expenseSourcesSchema = z.array(expenseSourceSchema);
+  const hasExpensesSchema = z.string().regex(/^true|false$/);
+  const formSchema = z.object({
+    hasExpenses: hasExpensesSchema,
+    expenses: expenseSourcesSchema,
+  });
+
+  const {
+    control,
+    formState: { errors },
+    handleSubmit,
+    watch,
+    getValues,
+  } = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      hasExpenses: formData.hasExpenses ? 'true' : 'false',
+      expenses: formData.expenses ?? [],
+    },
+  });
+  const watchHasExpenses = watch('hasExpenses');
+  // @ts-ignore
+  const hasTruthyExpenses = watchHasExpenses === 'true' || watchHasExpenses === true;
+  const { fields, append, remove, replace } = useFieldArray({
+    control,
+    name: 'expenses',
+  });
+
+  useEffect(() => {
+    const noExpensesAreListed = Number(getValues('expenses').length) === 0;
+    if (hasTruthyExpenses && noExpensesAreListed) {
+      append({
+        expenseSourceName: '',
+        expenseAmount: '',
+      });
+    }
+
+    if (!hasTruthyExpenses) {
+      replace([]);
+    }
+  }, [getValues('hasExpenses')]);
+
+  const formSubmitHandler: SubmitHandler<z.infer<typeof formSchema>> = async (expensesObject) => {
+    const hasExpensesBoolean: boolean = expensesObject.hasExpenses === 'true';
+    if (uuid) {
+      const updatedFormData = { ...formData, ...expensesObject, hasExpenses: hasExpensesBoolean };
+      setFormData(updatedFormData);
+      await updateScreen(uuid, updatedFormData, locale);
+      nextStep();
+    }
+  };
+
+  const createExpenseMenuItems = (expenseOptions: Record<string, FormattedMessageType>) => {
+    const disabledSelectMenuItem = (
+      <MenuItem value="select" key="disabled-select-value" disabled>
+        <FormattedMessage id="expenseBlock.createExpenseMenuItems-disabledSelectMenuItemText" defaultMessage="Select" />
+      </MenuItem>
+    );
+
+    const menuItemKeys = Object.keys(expenseOptions);
+    const menuItemLabels = Object.values(expenseOptions);
+
+    const menuItems = menuItemKeys.map((menuItemKey, i) => {
+      return (
+        <MenuItem value={menuItemKey} key={menuItemKey}>
+          {menuItemLabels[i]}
+        </MenuItem>
+      );
+    });
+
+    return [disabledSelectMenuItem, menuItems];
+  };
+
+  const getExpenseSourceLabel = (expenseOptions: Record<string, FormattedMessageType>, expenseSourceName: string) => {
+    if (expenseSourceName) {
+      return (
+        <>
+          {' ('}
+          {expenseOptions[expenseSourceName]}
+          {')'}?
+        </>
+      );
+    }
+
+    return '?';
```

Comment:
We should not return a question mark because not every language uses question marks. Maybe we can put the question mark before the parenthesis. i.e. It would look like "How much is this expense every month? (Rent)"